frontend/src/routes/(authenticated)/admin/+page.ts

```diff
@@ -139,6 +135,37 @@ export async function load(event: PageLoadEvent) {
   }
 }
 
+function buildUserSearchFilter(searchParams: URLSearchParams, user: LexAuthUser): UserFilterInput {
+  const userSearch = getSearchParam<AdminSearchParams>('userSearch', searchParams) ?? '';
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment -- false positive?
+  const userType = getSearchParam<AdminSearchParams, UserType>('userType', searchParams);
+  const onlyUsersICreated = getBoolSearchParam<AdminSearchParams>('usersICreated', searchParams);
+
+  const userSearchFilter: UserFilterInput =
+    isGuid(userSearch) ? {id: {eq: userSearch}}
+      : {
+        or: [
+          {name: {icontains: userSearch}},
+          {email: {icontains: userSearch}},
+          {username: {icontains: userSearch}}
+        ]
+      };
+  const usersICreatedFilter: UserFilterInput =
+    onlyUsersICreated ? {createdById: {eq: user.id as UUID}}
+      : {};
+  const userTypeFilter: UserFilterInput =
+    userType === 'admin' ? {isAdmin: {eq: true}}
+      : userType === 'nonAdmin' ? {isAdmin: {eq: false}}
+        : userType === 'guest' ? {createdById: {neq: null}}
+          : {};
```

Comment:
could this be a switch statement instead?

Reply:
I rewrote it. Among other things, I used a switch for this.